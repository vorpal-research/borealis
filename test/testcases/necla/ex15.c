#include "defines.h"

typedef struct {
   int* a;
   int* b;
   int* c;
} ptrStruct;

int i;

// @requires \is_valid_ptr(ptr)
// @requires \is_valid_ptr(ptr->a)
// @requires \is_valid_ptr(ptr->b)
int foo(ptrStruct* ptr) {
   *(ptr->a) = i++;
   *(ptr->b) = i++;
   return 1;
}

int main() {
   int x,y,z;
   int a,b,c;
   ptrStruct A,B;

   i = 0;

   A.a = &x;
   A.b = &y;
   A.c = &z;
   foo(&A);

   B.a = &a;
   B.b = &b;
   B.c = &c;
   foo(&B);
   
   ASSERT( *(B.b) >= *(A.a) );

   return -1;
}
