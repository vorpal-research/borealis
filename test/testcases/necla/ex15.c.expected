
[
	
	{
		"location" : 
		{
			"filename" : "test/testcases/necla/ex15.c",
			"loc" : 
			{
				"col" : 4,
				"line" : 13
			}
		},
		"type" : "INI-03"
	},
	
	{
		"location" : 
		{
			"filename" : "test/testcases/necla/ex15.c",
			"loc" : 
			{
				"col" : 4,
				"line" : 14
			}
		},
		"type" : "INI-03"
	},
	
	{
		"location" : 
		{
			"filename" : "test/testcases/necla/ex15.c",
			"loc" : 
			{
				"col" : 4,
				"line" : 35
			}
		},
		"type" : "REQ-01"
	}
]
